pkg/mimirtool/client/client.go

```diff
@@ -86,7 +88,7 @@ func New(cfg Config) (*MimirClient, error) {
 		client = http.Client{Transport: transport}
 	}
 
-	path := rulerAPIPath
+	path := cfg.APIPrefix + rulerAPIPath
```

Comment:
Can you use `path.Join`? that will make it not necessary to think about the slashes in the flags

Reply:
Can you also add some validation on `--use-legacy-routes` and `--api-prefix`? I imagine an error when both are set, since legacy-routes invalidates api-prefix.